### PR TITLE
Add documentation for logging in odo and odo create enhancements

### LINKED ISSUE #667: Add documentation about the logging in odo using glog
Add documentation about the logging in odo using glog

### PR DESCRIPTION
**Pull request by anmolbabu**
This PR adds the following information:

1. Enhancements in odo create command added to docs/cli-reference.md
2. docs/logging.md describes about the use of glog for logging

fixes #667
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

### CODE REVIEW

File: docs/logging.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Logging in ODO
+
+[Glog](https://godoc.org/github.com/golang/glog) is used for v style logging in ODO.
+glog provides functions such as Info, Warning, Error, Fatal, plus formatting variants such as Infof. It also provides V-style logging controlled by the -v for better runtime control
```
[the remark this answers is not in the document]
Comment (anmolbabu) [reply docs/logging.md]: 0 to 9 as in the link in https://github.com/redhat-developer/odo/pull/673/files#diff-6fede6782d33a3306a99cb1a47ea5f6bR28

[the remark this answers is not in the document]
Comment (anmolbabu) [reply docs/logging.md]: added the same here now

File: cmd/root.go [hunk 2]
```diff
@@ -122,6 +122,10 @@ func init() {
 	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 	pflag.CommandLine.Set("logtostderr", "true")
 
+	// Override the verbosity flag description
+	verbosity := pflag.Lookup("v")
+	verbosity.Usage += ". Level varies from 0 to 9 (default 0)."
```
Comment (cdrage) [inline cmd/root.go]: :+1:

File: docs/logging.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# Logging in Odo
+
+[Glog](https://godoc.org/github.com/golang/glog) is used for V style logging in Odo.
+
+
+## Working
+
+Every ODO command takes an optional flag `-v` that must be used with an integer log level in the range from 0-9. And any INFO severity log statement that is logged at a level lesser than or equal to the one passed with the command alongside `-v` flag will be logged to the STDOUT.
```
Comment (cdrage) [inline docs/logging.md]: Should be `Odo` not ODO.

`Any INFO severity log statement` Don't start sentences with "And"

`logged to STDOUT`

File: docs/logging.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+All ERROR severity level log messages will always be logged immaterial of the passed `v` level.
```
Comment (cdrage) [inline docs/logging.md]: `always be logged regardless of the passed V level.`

File: docs/logging.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+
+## Usage
+
+Every source file that needs to log messages will need to import glog as under:
+
+```
```
Comment (cdrage) [inline docs/logging.md]: make it ```go

not ```

so there is formatting in markdown

File: docs/logging.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+## Working
+
+Every ODO command takes an optional flag `-v` that must be used with an integer log level in the range from 0-9. And any INFO severity log statement that is logged at a level lesser than or equal to the one passed with the command alongside `-v` flag will be logged to the STDOUT.
+
+All ERROR severity level log messages will always be logged immaterial of the passed `v` level.
+
+
+## Usage
+
+Every source file that needs to log messages will need to import glog as under:
```
Comment (cdrage) [inline docs/logging.md]: Every source file that requires logging will need to import `glog`

File: docs/logging.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import "github.com/golang/glog"
+```
+
+Any default debug level severity messages need to be logged using:
+
+```
+glog.V(4).Infof(msg, args...)
+```
+
+For more info level logging conventions please refer [here](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#kubectl-output-verbosity-and-debugging)
```
Comment (cdrage) [inline docs/logging.md]: missing period at end of sentence

File: docs/logging.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+```
+glog.Errorf(msg, args...)
+```
+
+Warning messages can be logged as under:
+
+```
+glog.Warningf(msg, args...)
+```
+Similarly there are also Info, Error and Warning counterparts of each of thes above that don't take format args but instead a string message
```
Comment (cdrage) [inline docs/logging.md]: missing newline above

Comment (cdrage) [reply docs/logging.md]: `Similarly there is also`

Comment (cdrage) [reply docs/logging.md]: `these`

Comment (cdrage) [reply docs/logging.md]: Actually, maybe remove this sentence. The user can look at https://kubernetes.io/docs/reference/kubectl/cheatsheet/#kubectl-output-verbosity-and-debugging if they need help

File: docs/logging.md (hunk 3 above)
Comment (anmolbabu) [reply docs/logging.md]: Done

File: docs/logging.md (hunk 4 above)
Comment (anmolbabu) [reply docs/logging.md]: Done

File: docs/logging.md (hunk 5 above)
Comment (anmolbabu) [reply docs/logging.md]: Did you mean get ``` and the statement in same line ?
If so, done

File: docs/logging.md (hunk 6 above)
Comment (anmolbabu) [reply docs/logging.md]: Done

File: docs/logging.md (hunk 7 above)
Comment (anmolbabu) [reply docs/logging.md]: Done

File: docs/logging.md (hunk 8 above)
Comment (anmolbabu) [reply docs/logging.md]: Done

**Comment by cdrage** [comment]
This LGTM! Let's merge.

**Pull request merged**